pylint/checkers/utils.py

```diff
@@ -770,65 +767,65 @@ def decorated_with_property(node: astroid.FunctionDef) -> bool:
+    if not isinstance(node, (astroid.UnboundMethod, nodes.FunctionDef)):
         return False
     if node.decorators:
         for decorator in node.decorators.nodes:
-            if isinstance(decorator, astroid.Attribute) and decorator.attrname in kinds:
+            if isinstance(decorator, nodes.Attribute) and decorator.attrname in kinds:
                 return True
     return False
 
 
-def is_property_setter(node: astroid.FunctionDef) -> bool:
+def is_property_setter(node: nodes.FunctionDef) -> bool:
     """Check if the given node is a property setter"""
     return _is_property_kind(node, "setter")
 
 
-def is_property_deleter(node: astroid.FunctionDef) -> bool:
+def is_property_deleter(node: nodes.FunctionDef) -> bool:
     """Check if the given node is a property deleter"""
     return _is_property_kind(node, "deleter")
 
 
-def is_property_setter_or_deleter(node: astroid.FunctionDef) -> bool:
+def is_property_setter_or_deleter(node: nodes.FunctionDef) -> bool:
     """Check if the given node is either a property setter or a deleter"""
     return _is_property_kind(node, "setter", "deleter")
 
 
-def _is_property_decorator(decorator: astroid.Name) -> bool:
+def _is_property_decorator(decorator: nodes.Name) -> bool:
     for inferred in decorator.infer():
-        if isinstance(inferred, astroid.ClassDef):
+        if isinstance(inferred, nodes.ClassDef):
             if inferred.qname() in ("builtins.property", "functools.cached_property"):
                 return True
             for ancestor in inferred.ancestors():
                 if ancestor.name == "property" and ancestor.root().name == BUILTINS:
                     return True
-        elif isinstance(inferred, astroid.FunctionDef):
+        elif isinstance(inferred, nodes.FunctionDef):
             # If decorator is function, check if it has exactly one return
             # and the return is itself a function decorated with property
-            returns: List[astroid.Return] = list(
+            returns: List[nodes.Return] = list(
                 inferred._get_return_nodes_skip_functions()
             )
             if len(returns) == 1 and isinstance(
-                returns[0].value, (astroid.Name, astroid.Attribute)
+                returns[0].value, (nodes.Name, nodes.Attribute)
             ):
                 inferred = safe_infer(returns[0].value)
                 if (
                     inferred
                     and isinstance(inferred, astroid.objects.Property)
-                    and isinstance(inferred.function, astroid.FunctionDef)
+                    and isinstance(inferred.function, nodes.FunctionDef)
                 ):
                     return decorated_with_property(inferred.function)
     return False
 
 
 def decorated_with(
-    func: Union[astroid.FunctionDef, astroid.BoundMethod, astroid.UnboundMethod],
+    func: Union[nodes.FunctionDef, astroid.BoundMethod, astroid.UnboundMethod],
```

Comment:
I was suspicious that astroid.bases was containing nodes too.. `BoundMethod` and `UnboundMethod` could be considered like nodes, right ?

Reply:
`BoundMethod` and `UnboundMethod` inherit from `Proxy` not `NodeNG`. So I would say no.